From bef63ab5c58e2e74e129518850532ad58d354809 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Wed, 25 Sep 2024 16:03:13 +0800
Subject: [PATCH 2/5] Compatible with libirecovery static library mode

---
 src/idevicerestore.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/idevicerestore.c b/src/idevicerestore.c
index bf58a3e..574dd7c 100644
--- a/src/idevicerestore.c
+++ b/src/idevicerestore.c
@@ -1921,12 +1921,21 @@ int main(int argc, char* argv[]) {
 
 	curl_global_init(CURL_GLOBAL_ALL);
 
+	// need manual initialization of libirecovery when using static lib
+#ifdef IRECV_STATIC
+	irecv_init();
+#endif
+
 	result = idevicerestore_start(client);
 
 	idevicerestore_client_free(client);
 
 	curl_global_cleanup();
 
+#ifdef IRECV_STATIC
+	irecv_exit();
+#endif
+
 	return (result == 0) ? EXIT_SUCCESS : EXIT_FAILURE;
 }
 #endif
-- 
2.40.1.windows.1

