From f21eab00d805bccd243b3890f76d48c156b7cd37 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Tue, 5 Nov 2024 17:59:34 +0800
Subject: [PATCH 2/6] Compatible with libirecovery static library mode

---
 src/idevicerestore.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/idevicerestore.c b/src/idevicerestore.c
index b8bb1d0..1b6483d 100644
--- a/src/idevicerestore.c
+++ b/src/idevicerestore.c
@@ -1955,12 +1955,21 @@ int main(int argc, char* argv[]) {
 
 	curl_global_init(CURL_GLOBAL_ALL);
 
+	// need manual initialization of libirecovery when using static lib
+#ifdef IRECV_STATIC
+	irecv_init();
+#endif
+
 	result = idevicerestore_start(client);
 
 	idevicerestore_client_free(client);
 
 	curl_global_cleanup();
 
+#ifdef IRECV_STATIC
+	irecv_exit();
+#endif
+
 	return (result == 0) ? EXIT_SUCCESS : EXIT_FAILURE;
 }
 #endif
-- 
2.40.1.windows.1

