From dfd92583712a6a9a554a0a15803ea865c980c778 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Wed, 25 Sep 2024 16:03:13 +0800
Subject: [PATCH 3/5] Fixed stat failed in win32 large files

---
 src/ipsw.c | 15 +++++++++++++++
 src/ipsw.h |  5 +++++
 2 files changed, 20 insertions(+)

diff --git a/src/ipsw.c b/src/ipsw.c
index da7528d..2cb8929 100644
--- a/src/ipsw.c
+++ b/src/ipsw.c
@@ -45,6 +45,13 @@
 #include "common.h"
 #include "idevicerestore.h"
 
+#ifdef WIN32
+#	define stat _stat64
+#	define size_t unsigned __int64
+#else
+#	define stat stat
+#endif
+
 #define BUFSIZE 0x100000
 
 static int cancel_flag = 0;
@@ -1500,3 +1507,11 @@ int64_t ipsw_file_tell(ipsw_file_handle_t handle)
 		return -1;
 	}
 }
+
+#ifdef stat
+#	undef stat
+#endif
+
+#ifdef size_t
+#	undef size_t
+#endif
diff --git a/src/ipsw.h b/src/ipsw.h
index 8cb2561..4c5fc17 100644
--- a/src/ipsw.h
+++ b/src/ipsw.h
@@ -43,8 +43,13 @@ void ipsw_close(ipsw_archive_t ipsw);
 
 int ipsw_print_info(const char* ipsw);
 
+#ifdef WIN32
+typedef int (*ipsw_list_cb)(void *ctx, ipsw_archive_t ipsw, const char *name, struct _stat64*stat);
+typedef int (*ipsw_send_cb)(void *ctx, void *data, uint64_t size, uint64_t done, uint64_t total_size);
+#else
 typedef int (*ipsw_list_cb)(void *ctx, ipsw_archive_t ipsw, const char *name, struct stat *stat);
 typedef int (*ipsw_send_cb)(void *ctx, void *data, size_t size, size_t done, size_t total_size);
+#endif
 
 struct ipsw_file_handle {
 	FILE* file;
-- 
2.40.1.windows.1

