From 23ec60691630cae9f1dd491529dd36c67e035992 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Thu, 25 Sep 2025 16:17:11 +0800
Subject: [PATCH 4/8] Fix ftab.c compilation issues in msvc

---
 src/ftab.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ftab.c b/src/ftab.c
index d0fc26b..07d7ab9 100644
--- a/src/ftab.c
+++ b/src/ftab.c
@@ -28,7 +28,7 @@
 #include "common.h"
 #include "endianness.h"
 
-int ftab_parse(const void *data, size_t data_size, ftab_t *ftab, uint32_t *tag)
+int ftab_parse(const char *data, size_t data_size, ftab_t *ftab, uint32_t *tag)
 {
 	if (!data || !data_size || !ftab) {
 		return -1;
-- 
2.40.1.windows.1

