From bed364cb166364bbb22844ad13ca78339c46c52e Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Thu, 25 Sep 2025 16:17:28 +0800
Subject: [PATCH 5/8] Fix mbn.h compilation issues in msvc

---
 src/mbn.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mbn.h b/src/mbn.h
index a328128..86731d9 100644
--- a/src/mbn.h
+++ b/src/mbn.h
@@ -96,7 +96,7 @@ typedef struct {
 	} header;
 	uint32_t parsed_size;
 	uint32_t parsed_sig_offset;
-	void* data;
+	char* data;
 	uint32_t size;
 } mbn_file;
 
-- 
2.40.1.windows.1

