From 9b1bf06befe93eef8bc44e001f37055141093fc2 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Thu, 25 Jul 2024 09:38:12 +0800
Subject: [PATCH 5/5] Fixed Error C2016

---
 src/restore.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/restore.c b/src/restore.c
index d3828f9..de55795 100644
--- a/src/restore.c
+++ b/src/restore.c
@@ -636,9 +636,11 @@ const char* restore_progress_string(unsigned int operation)
 	}
 }
 
+#ifndef WIN32
 struct restored_service_client {
 
 };
+#endif
 
 #define SERVICE_TYPE_RESTORED 1
 #define SERVICE_TYPE_PLIST 2
-- 
2.40.1.windows.1

