From 3575e10ad68f31758a5056ed610e16de885babdc Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Wed, 23 Apr 2025 14:18:41 +0800
Subject: [PATCH 5/5] opt errno is set when optimizing ipsw open() fails

---
 src/ipsw.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ipsw.c b/src/ipsw.c
index fd0d5a3..63f5eef 100644
--- a/src/ipsw.c
+++ b/src/ipsw.c
@@ -372,6 +372,7 @@ ipsw_archive_t ipsw_open(const char* ipsw)
 		if (zip == NULL) {
 			error("ERROR: zip_open: %s: %d\n", ipsw, err);
 			free(archive);
+			_set_errno(err);
 			return NULL;
 		}
 		archive->zip = 1;
-- 
2.40.1.windows.1

