From 079463beee224c649764fb0aac6bd2fde13bcae9 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Thu, 25 Sep 2025 16:19:41 +0800
Subject: [PATCH 7/8] Fix ipsw.c utf-8 filename bug in msvc

---
 src/ipsw.c | 61 ++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 61 insertions(+)

diff --git a/src/ipsw.c b/src/ipsw.c
index cebfa49..2049f40 100644
--- a/src/ipsw.c
+++ b/src/ipsw.c
@@ -45,6 +45,58 @@
 #include "common.h"
 #include "idevicerestore.h"
 
+
+#ifdef WIN32
+inline const uintptr_t _utf82utf16(const char* input, wchar_t* output)
+{
+	int len = MultiByteToWideChar(CP_UTF8, 0, input, -1, NULL, 0);
+    if (!output)
+        return (uintptr_t)len;
+    MultiByteToWideChar(CP_UTF8, 0, input, -1, (LPWSTR)output, len);
+    return (uintptr_t)output;
+}
+
+struct _stat64_utf8
+{
+    _dev_t         st_dev;
+    _ino_t         st_ino;
+    unsigned short st_mode;
+    short          st_nlink;
+    short          st_uid;
+    short          st_gid;
+    _dev_t         st_rdev;
+    __int64        st_size;
+    __time64_t     st_atime;
+    __time64_t     st_mtime;
+    __time64_t     st_ctime;
+};
+
+inline int _stat64_utf8(
+	char const* const path, struct _stat64* result)
+{
+	int count, size;
+	wchar_t* wbuf = NULL;
+
+	count = _utf82utf16(path, 0);
+	size = (count + 1) * 2;
+	wbuf = malloc(size);
+	memset(wbuf, 0, size);
+	_utf82utf16(path, wbuf);
+
+	result = _wstat64(wbuf, result);
+	free(wbuf);
+
+	return result;
+}
+#endif
+
+#ifdef WIN32
+#	define stat _stat64_utf8
+#	define size_t unsigned __int64
+#else
+#	define stat stat
+#endif
+
 #define BUFSIZE 0x100000
 
 static int cancel_flag = 0;
@@ -318,6 +370,7 @@ ipsw_archive_t ipsw_open(const char* ipsw)
 		if (zip == NULL) {
 			logger(LL_ERROR, "zip_open: %s: %d\n", ipsw, err);
 			free(archive);
+			_set_errno(err);
 			return NULL;
 		}
 		archive->zip = 1;
@@ -1540,3 +1593,11 @@ int64_t ipsw_file_tell(ipsw_file_handle_t handle)
 		return -1;
 	}
 }
+
+#ifdef stat
+#	undef stat
+#endif
+
+#ifdef size_t
+#	undef size_t
+#endif
\ No newline at end of file
-- 
2.40.1.windows.1

