From 7396c7efbbf1bacb732a08ce19eb19f9fe35b624 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Thu, 25 Sep 2025 16:20:38 +0800
Subject: [PATCH 8/8] Fix ipsw large files bug in win32

---
 src/ipsw.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/ipsw.h b/src/ipsw.h
index b22e614..346cca8 100644
--- a/src/ipsw.h
+++ b/src/ipsw.h
@@ -43,8 +43,13 @@ void ipsw_close(ipsw_archive_t ipsw);
 
 int ipsw_print_info(const char* ipsw);
 
+#ifdef WIN32
+typedef int (*ipsw_list_cb)(void *ctx, ipsw_archive_t ipsw, const char *name, struct _stat64*stat);
+typedef int (*ipsw_send_cb)(void *ctx, void *data, uint64_t size, uint64_t done, uint64_t total_size);
+#else
 typedef int (*ipsw_list_cb)(void *ctx, ipsw_archive_t ipsw, const char *name, struct stat *stat);
-typedef int (*ipsw_send_cb)(void *ctx, const void *data, size_t size, size_t done, size_t total_size);
+typedef int (*ipsw_send_cb)(void *ctx, void *data, size_t size, size_t done, size_t total_size);
+#endif
 
 struct ipsw_file_handle {
 	FILE* file;
-- 
2.40.1.windows.1

