From 239a6cb5fb2ca482dcd3850dc9555d4b0c5f8485 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Wed, 25 Sep 2024 16:03:15 +0800
Subject: [PATCH 1/2] [PATCH] 001_fix_static_build

---
 src/libirecovery.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libirecovery.c b/src/libirecovery.c
index ddc1946..ac243a7 100644
--- a/src/libirecovery.c
+++ b/src/libirecovery.c
@@ -635,6 +635,8 @@ static thread_once_t deinit_once = THREAD_ONCE_INIT;
   #endif
 #endif
 
+#ifndef IRECV_STATIC  // disable dll constructor
+
 #ifdef HAVE_ATTRIBUTE_CONSTRUCTOR
 static void __attribute__((constructor)) libirecovery_initialize(void)
 {
@@ -664,6 +666,8 @@ BOOL WINAPI DllMain(HINSTANCE hModule, DWORD dwReason, LPVOID lpReserved)
 #warning No compiler support for constructor/destructor attributes, some features might not be available.
 #endif
 
+#endif  // disable dll constructor
+
 #ifdef HAVE_IOKIT
 static int iokit_get_string_descriptor_ascii(irecv_client_t client, uint8_t desc_index, unsigned char * buffer, int size)
 {
-- 
2.40.1.windows.1

