From 5f4dd0b40155908dac6e550c4b03bb150ebad909 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Tue, 5 Nov 2024 17:59:36 +0800
Subject: [PATCH 1/2] [PATCH] 001_fix_static_build

---
 src/libirecovery.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/libirecovery.c b/src/libirecovery.c
index e8e2cab..8e4b2f3 100644
--- a/src/libirecovery.c
+++ b/src/libirecovery.c
@@ -646,6 +646,8 @@ static thread_once_t deinit_once = THREAD_ONCE_INIT;
   #endif
 #endif
 
+#ifndef IRECV_STATIC  // disable dll constructor
+
 #ifdef HAVE_ATTRIBUTE_CONSTRUCTOR
 static void __attribute__((constructor)) libirecovery_initialize(void)
 {
@@ -675,6 +677,8 @@ BOOL WINAPI DllMain(HINSTANCE hModule, DWORD dwReason, LPVOID lpReserved)
 #warning No compiler support for constructor/destructor attributes, some features might not be available.
 #endif
 
+#endif  // disable dll constructor
+
 #ifdef HAVE_IOKIT
 static int iokit_get_string_descriptor_ascii(irecv_client_t client, uint8_t desc_index, unsigned char * buffer, int size)
 {
-- 
2.40.1.windows.1

