From 1384e1b0fd54b3d102365382d86b757e65631493 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Tue, 5 Nov 2024 17:59:37 +0800
Subject: [PATCH 2/2] Fixed usleep() failing to compile under win

---
 src/libirecovery.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libirecovery.c b/src/libirecovery.c
index 8e4b2f3..a56bccd 100644
--- a/src/libirecovery.c
+++ b/src/libirecovery.c
@@ -2457,7 +2457,7 @@ static void* _irecv_handle_device_add(void *userdata)
 				break;
 			}
 			debug("Failed to set configuration, error %d, retrying...\n", error);
-			usleep(500000);
+			sleep(500000);
 		}
 		if (error != IRECV_E_SUCCESS) {
 			debug("Failed to set configuration, error %d\n", error);
@@ -2471,7 +2471,7 @@ static void* _irecv_handle_device_add(void *userdata)
 				break;
 			}
 			debug("Failed to set interface, error %d, retrying...\n", error);
-			usleep(500000);
+			sleep(500000);
 		}
 		if (error != IRECV_E_SUCCESS) {
 			debug("Failed to set interface, error %d\n", error);
-- 
2.40.1.windows.1

