From 1ab3d91214af8d88790121e81ba6640fb9f0432a Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Wed, 25 Sep 2024 16:19:41 +0800
Subject: [PATCH 2/2] Fixed usleep() failing to compile under win

---
 src/libirecovery.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libirecovery.c b/src/libirecovery.c
index ac243a7..f8090ae 100644
--- a/src/libirecovery.c
+++ b/src/libirecovery.c
@@ -2446,7 +2446,7 @@ static void* _irecv_handle_device_add(void *userdata)
 				break;
 			}
 			debug("Failed to set configuration, error %d, retrying...\n", error);
-			usleep(500000);
+			sleep(500000);
 		}
 		if (error != IRECV_E_SUCCESS) {
 			debug("Failed to set configuration, error %d\n", error);
@@ -2460,7 +2460,7 @@ static void* _irecv_handle_device_add(void *userdata)
 				break;
 			}
 			debug("Failed to set interface, error %d, retrying...\n", error);
-			usleep(500000);
+			sleep(500000);
 		}
 		if (error != IRECV_E_SUCCESS) {
 			debug("Failed to set interface, error %d\n", error);
-- 
2.40.1.windows.1

