diff --git a/src/libirecovery.c b/src/libirecovery.c
index a60782c..514d851 100644
--- a/src/libirecovery.c
+++ b/src/libirecovery.c
@@ -619,6 +619,8 @@ static thread_once_t deinit_once = THREAD_ONCE_INIT;
   #endif
 #endif
 
+#ifndef IRECV_STATIC  // disable dll constructor
+
 #ifdef HAVE_ATTRIBUTE_CONSTRUCTOR
 static void __attribute__((constructor)) libirecovery_initialize(void)
 {
@@ -648,6 +650,8 @@ BOOL WINAPI DllMain(HINSTANCE hModule, DWORD dwReason, LPVOID lpReserved)
 #warning No compiler support for constructor/destructor attributes, some features might not be available.
 #endif
 
+#endif  // disable dll constructor
+
 #ifdef HAVE_IOKIT
 static int iokit_get_string_descriptor_ascii(irecv_client_t client, uint8_t desc_index, unsigned char * buffer, int size)
 {
