From 2656c7a6be744fd93124f33d8e090d951604f3c2 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Tue, 5 Nov 2024 17:59:35 +0800
Subject: [PATCH 2/2] fix ideviceprovision compilation problem

---
 tools/ideviceprovision.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/ideviceprovision.c b/tools/ideviceprovision.c
index 4080a28..9de7593 100644
--- a/tools/ideviceprovision.c
+++ b/tools/ideviceprovision.c
@@ -38,6 +38,7 @@
 
 #ifdef WIN32
 #include <windows.h>
+#include <WinSock2.h>
 #else
 #include <arpa/inet.h>
 #endif
-- 
2.40.1.windows.1

