From 1b6c2dd5ab5f4c626d2a866acd24e57e8f98a202 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Tue, 5 Nov 2024 17:59:38 +0800
Subject: [PATCH 3/4] 003_fix_build_dll

---
 include/plist/plist.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/plist/plist.h b/include/plist/plist.h
index aff81e9..4724c09 100644
--- a/include/plist/plist.h
+++ b/include/plist/plist.h
@@ -78,6 +78,8 @@ extern "C"
 #ifndef PLIST_API
   #ifdef LIBPLIST_STATIC
     #define PLIST_API
+  #elif defined(PLIST_EXPORT)
+    #define PLIST_API __declspec(dllexport)
   #elif defined(_WIN32)
     #define PLIST_API __declspec(dllimport)
   #else
-- 
2.40.1.windows.1

