From aa7fddf448bf4a73045b7db647c5bc318cef9dd4 Mon Sep 17 00:00:00 2001
From: zero <zero.kwok@foxmail.com>
Date: Wed, 24 Jul 2024 17:41:24 +0800
Subject: [PATCH 3/3] 003_fix_tools_msvc

---
 tools/inetcat.c | 9 +++++++++
 tools/iproxy.c  | 2 ++
 2 files changed, 11 insertions(+)

diff --git a/tools/inetcat.c b/tools/inetcat.c
index f70215b..748db15 100644
--- a/tools/inetcat.c
+++ b/tools/inetcat.c
@@ -33,7 +33,9 @@
 #include <string.h>
 #include <fcntl.h>
 #include <stddef.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#endif
 #include <errno.h>
 #include <getopt.h>
 #ifdef WIN32
@@ -50,6 +52,13 @@
 #include "usbmuxd.h"
 #include <libimobiledevice-glue/socket.h>
 
+#ifdef _MSC_VER
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#define STDIN_FILENO  _fileno(stdin)
+#define STDOUT_FILENO _fileno(stdout)
+#endif
+
 static int debug_level = 0;
 
 static size_t read_data_socket(int fd, uint8_t* buf, size_t bufsize)
diff --git a/tools/iproxy.c b/tools/iproxy.c
index 1bb9e77..0f71fc9 100644
--- a/tools/iproxy.c
+++ b/tools/iproxy.c
@@ -34,7 +34,9 @@
 #include <string.h>
 #include <fcntl.h>
 #include <stddef.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#endif
 #include <errno.h>
 #include <getopt.h>
 #ifdef WIN32
-- 
2.40.1.windows.1

